src/Psalm/Provider/FileStorageProvider.php

```diff
@@ -31,6 +31,14 @@ public function __construct(FileStorageCacheProvider $cache)
     public function get($file_path)
     {
         $file_path = strtolower($file_path);
+        $temp_fp = $file_path;
+
+        if(strpos($file_path, '/./') !== false) {
+            $pos = strpos($file_path, '/./');
+            $temp_fp = substr($file_path,0,$pos).substr($file_path,$pos+2);
+        }
+        
+        $file_path=$temp_fp;
```

Comment:
How about just:

```php
if strpos($file_path, '/./') {
    $file_path = str_replace($file_path, '/./', '/')
}
```

I'm nitpicking because it'd be nice to make a PR upstream.

Reply:
Yes it is good

Reply:
Tenets from this: https://trello.com/c/zVMaIbu5/141-false https://trello.com/c/rc24YMir/142-use-strreplace https://trello.com/c/U6pnq4v0/143-unnecessary-temp